astropy/convolution/convolve.py

```diff
@@ -700,9 +700,6 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
         # need to re-zero weights outside of the image (if it is padded, we
         # still don't weight those regions)
         bigimwt[arrayslices] = wtsm.real[arrayslices]
-        # curiously, at the floating-point limit, can get slightly negative numbers
```

Comment:
The problem with numpy-dev is that this weight file gets a very slightly positive number in one of the test cases. Hence, the rewrite (I also moved it lower down since it seemed more logical there).